src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -631,6 +631,135 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+// Generate stub for ghash process  blocks.
+//
+// Arguments for generated stub:
+//      state:  R3_ARG1
+//      subkeyH:    R4_ARG2
+//      data: R5_ARG3
+//
+address generate_ghash_processBlocks() {
+  StubCodeMark mark(this, "StubRoutines", "ghash");
+  address start = __ function_entry();
+
+  // Registers for parameters
+  Register state = R3_ARG1;                     // long[] state
+  Register subkeyH = R4_ARG2;                   // long[] subH
+  Register data = R5_ARG3;                      // byte[] data
+  Register blocks = R6_ARG4;
+  Register temp1 = R8;
+  Register temp2 = R9;
+  Register temp3 = R10;
+  Register temp4 = R11;
+  Register align = data;
+  // Vector Registers
+  VectorRegister vH = VR0;
+  VectorSRegister vHS = VSR32;
+  VectorRegister vX = VR1;
+  VectorRegister vH_shift = VR2;
+  VectorRegister vTmp1 = VR3;
+  VectorRegister vTmp2 = VR4;
+  VectorRegister vSwappedH = VR5;
+  VectorRegister vTmp4 = VR6;
+  VectorRegister loadOrder = VR7;
+  VectorRegister vMSB = VR8;
+  VectorRegister vLowerH = VR9;
+  VectorRegister vHigherH = VR10;
+  VectorRegister vZero = VR11;
+  VectorRegister vConst1 = VR12;
+  VectorRegister vConst7 = VR13;
+  VectorRegister vConstC2 = VR14;
+  VectorRegister vTmp3 = VR16;
+  VectorRegister vTmp5 = VR17;
+  VectorRegister vTmp6 = VR18;
+  VectorRegister vTmp7 = VR19;
+  VectorRegister vHigh = VR20;
+  VectorRegister vLow = VR21;
+  VectorRegister vPerm = VR22;
+  VectorRegister vZero_Stored = VR23;
+  VectorSRegister vZero_StoredS = VSR55;
+  VectorRegister vMask = VR24;
+  VectorRegister vS = VR25;
+  VectorSRegister vXS = VSR33;
+  Label L_end, L_aligned;
+
+  static const unsigned char perm_pattern[16] __attribute__((aligned(16))) = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
+
+  __ li(temp1, 0xc2);
+  __ sldi(temp1, temp1, 56);
+  // Load the vector from memory into vConstC2
+  __ vxor(vConstC2, vConstC2, vConstC2);
+  __ mtvrd(vConstC2, temp1);
```

Comment:
is that `vxor` instruction really necessary? `mtvrd` will do overwrite any way. So why do we want to be sure that there is 0 in `vConstC2` ?